src/3D/3D_utilities.F90

```diff
@@ -58,14 +58,15 @@ subroutine calcGridRatio(N, nStart, nEnd, s0, S, ratio)
     !     -------
     !     ratio : real
     !         The computed grid ratio that satifies all the inputs.
-    use hypInput, only: fullDeltas
+    use hypInput, only: fullDeltas, growthRatios
     use precision
 
     implicit none
 
     ! Input Parameters
     integer(kind=intType), intent(in) :: N, nStart, nEnd
-    real(kind=realType), intent(in) :: s0, S
+    real(kind=realType), intent(in) :: s0
+    real(kind=realType), intent(inout) :: S
```

Comment:
Related to my previous comment. The way that this function `calcGridRatio` is currently used is to pass in the `marchDist`, which is an input parameter. This will then modify that value, which is not great.